apps/rush-lib/src/api/EnvironmentConfiguration.ts

```diff
@@ -275,11 +314,29 @@ export class EnvironmentConfiguration {
             break;
           }
 
-          case EnvironmentVariableNames.RUSH_BUILD_CACHE_WRITE_CREDENTIAL: {
+          case EnvironmentVariableNames.RUSH_BUILD_CACHE_CREDENTIAL: {
             EnvironmentConfiguration._buildCacheCredential = value;
             break;
           }
 
+          case EnvironmentVariableNames.RUSH_BUILD_CACHE_ENABLED: {
+            if (value === 'true') {
+              EnvironmentConfiguration._buildCacheEnabled = true;
+            } else if (value === 'false') {
+              EnvironmentConfiguration._buildCacheEnabled = false;
+            }
```

Comment:
We should probably error if it's set to an unexpected value.

Reply:
Many of Rush's command-line parameters are mapped to environment variables using the [environmentVariable API](https://rushstack.io/pages/api/ts-command-line.ibasecommandlinedefinition.environmentvariable/) from `ts-command-line`. 

For booleans, "The value must be `1` for true, or `0` for false, otherwise a validation error is reported. An empty string causes the environment variable to be ignored."  The parsing looks like this:

https://github.com/microsoft/rushstack/blob/aa5cb93e68bdc8aab65f3216e806d5f57997ed49/libraries/ts-command-line/src/parameters/CommandLineFlagParameter.ts#L46-L56

It seems counterintuitive for other Rush environment variables to follow different rules.

> We should probably error if it's set to an unexpected value.

I especially agree that it's confusing for something like `RUSH_BUILD_CACHE_ENABLED=yes` to be silently ignored.  

Generally Rush is very strict about these things.  For example, if you merely misspell the name `RUSH_BUILD_CACHE_ENABLE=1` you can expect an error like:

```
ERROR: The following environment variables were found with the "RUSH_" prefix, but they 
are not recognized by this version of Rush: RUSH_BUILD_CACHE_ENABLE
```

Reply:
🤔 The `0` and `1` design seems slightly less elegant compared to `true` or `false`.

However if you are writing scripts that read these values, `0` and `1` do have an advantage that you can compare them without worrying about `true` vs `True` vs `TRUE`.

Reply:
(Looks like `RUSH_ALLOW_UNSUPPORTED_NODEJS` (from PR https://github.com/microsoft/rushstack/pull/1408) also uses the `true` string.  We should probably fix that one as well.)

Reply:
Also [this blog post](https://blog.plover.com/prog/sh-flags.html) is rather unsettling. 😁

Reply:
I took a look at the `ts-command-line` stuff and it makes sense to me, but pulling it out or reusing it looks pretty difficult.

I've added a quick equivalent helper to the `EnvironmentConfiguration` class and converted my 2 variables and the previous 2 variables to use it, and I used the same message as the one in `ts-command-line` for (hopeful) consistency.